### DISCUSSION TITLE
How to upgrade pipx on Ubuntu

### DISCUSSION THREAD

**Discussion by ghost** [pypa/pipx#1380] [Q&A] [answered]
I'm checking how to use pipx as an universal way for Ansible installation - [Installing and upgrading Ansible with pipx](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#installing-and-upgrading-ansible-with-pipx).

I can [install pipx on Ubuntu](https://pipx.pypa.io/stable/installation/#on-linux)
```shell
apt update
apt install pipx

# 22.04
pipx --version
1.0.0

# 24.04
pipx --version
1.4.3
```

And now, it is not clear how to [upgrade pipx](https://pipx.pypa.io/stable/installation/#upgrade-pipx)
```shell
apt upgrade pipx

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
pipx is already the newest version (1.4.3-1).
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

**Comment by huxuan** [comment, marked as answer]
I would recommend to install pipx with pipx, though it is not recommended in the documentation.

I started another discussion about the exact limitation, and that is actually how I use/upgrade pipx on ubuntu.

https://github.com/pypa/pipx/discussions/1341

**Comment by ghost** [comment]
Thanks, I also read that note, but looks like it may be a way to get the latest version of the app

[Install pipx](https://github.com/pypa/pipx#install-pipx)
<img width="828" alt="Screenshot 2024-04-29 at 02 16 24" src="https://github.com/pypa/pipx/assets/88528265/6cd72b12-7d96-4e63-9ad0-3475e87bcf26">

**Reply by huxuan** [reply]
IMO, there might be some historical problems, but I did not realize anything now. I also submitted an enhancement here, hope it will be released in the next version. https://github.com/pypa/pipx/pull/1340

**Comment by zinc75** [comment]
Can now be easily solved using https://github.com/pypa/pipx/issues/1481#issuecomment-2583365034 and https://github.com/zinc75/pipx-1.6.0-backport-ubuntu-2404-lts , at least for 24.04 . 

I've put the recipe for backporting newer versions of Ubuntu packages for pipx , which may be applicable also for 22.04 .
